NetworthManager.js

```diff
@@ -0,0 +1,207 @@
+        const prices = await parsePrices(prePrices, this.cachePrices, this.pricesRetries);
+        const items = await parseItems(profileData, museumData);
+        await this.itemsPromise.catch(() => {});
+        return calculateNetworth(items, purse, bankBalance, personalBankBalance, prices, this);
+    }
+
+    /**
+     * Returns the networth of a profile using pre-decoded items
+     * @param {object} profileData - The profile player data from the Hypixel API (profile.members[uuid])
+     * @param {{
+     *          armor: [],
+     *          equipment: [],
+     *          wardrobe: [],
+     *          inventory: [],
+     *          enderchest: [],
+     *          accessories: [],
+     *          personal_vault: [],
+     *          storage: [],
+     *          fishing_bag: [],
+     *          potion_bag: [],
+     *          sacks_bag: [],
+     *          candy_inventory: [],
+     *          carnival_mask_inventory: [],
+     *          museum: [],
+     *        }} items - Pre-parsed inventories, most inventories are just decoded except for sacks, essence, and pets which are parsed specifically as listed above, museum is an array of member[uuid].items and member[uuid].special combined and decoded (see parseItems.js)
+     * @param {number} bankBalance - The bank balance of the player from the Hypixel API (profile.banking.balance)
+     * @param {object} [prices] - A prices object generated from the getPrices function. If not provided, the prices will be retrieved every time the function is called
+     * @returns An object containing the player's networth calculation
+     */
+    async getPreDecodedNetworth({ profileData, items, bankBalance, prices: prePrices }) {
+        const purse = profileData.currencies?.coin_purse;
+        const personalBankBalance = profileData.profile?.bank_account;
+        await postParseItems(profileData, items);
+        const prices = await parsePrices(prePrices, this.cachePrices, this.pricesRetries);
+        await this.itemsPromise.catch(() => {});
+        return calculateNetworth(items, purse, bankBalance, personalBankBalance, prices, this);
+    }
+
+    /**
+     * Returns the networth of an item
+     * @param {object} item - The item the networth should be calculated for
+     * @param {object} [prices] - A prices object generated from the getPrices function. If not provided, the prices will be retrieved every time the function
+     * @returns {object} - An object containing the item's networth calculation
+     */
+    async getItemNetworth({ item, prices: prePrices }) {
+        if (item?.tag === undefined && item?.exp === undefined) throw new NetworthError('Invalid item provided');
+        const prices = await parsePrices(prePrices, this.cachePrices, this.pricesRetries);
+        await this.itemsPromise;
+        return calculateItemNetworth(item, prices, this);
+    }
+}
+
+async function updateItems(retries = 3, retryInterval = 1000, currentRetry = 0) {
```

Comment:
Might as well stick these methods in the class. You can make them class methods as well so that people can run them with `NetworthManager.updateItems()` or `NetworthManager().updateItems()`.

Reply:
good point